lib/services/name/name-service.ts

```diff
@@ -0,0 +1,36 @@
+import { RpcConsummer } from '../../modules/rpc-consumer/rpc-consumer';
+import { NonaUserError } from '../../shared/errors/user-error';
+import {
+  NanoAddress,
+  NanoTarget,
+  NanoUsername,
+  isValidNanoAddress,
+  isValidNanoUsername,
+} from '../../shared/utils/address';
+import { KeyService } from '../hash/key-service';
+import { Rpc } from '../rpc/rpc';
+import { NameInfo } from './name-schemas';
+
+export class NameService extends RpcConsummer {
+  constructor() {
+    super(new Rpc({ url: 'http://rpc.nano.to' }));
+  }
+  public async resolveUsername(username: NanoUsername): Promise<NanoAddress> {
+    const res = await this.rpc.call('account_key', { account: username });
```

Comment:
We need to deal with the case where the `username` is not registered.
This is a bit tricky because the `rpc.nano.to` error message format is different than a standard rpc error message, so you may need to change the rpc error handler.

Alternatively, rather than using the public rpc of `rpc.nano.to` which is limited to 30 calls per minute without an api key, you can use the `https://nano.to/.well-known/nano-currency.json?names=username` endpoint referenced in the [README](https://github.com/fwd/nano-to/tree/master?tab=readme-ov-file#developer-api). But the answer format is different and you will need to create a new axios call.

Both solutions are fine in my opinion, implement it as you wish.

Reply:
I started off with using the .well-known endpoint of nano.to. It also works, but for a clean implementation, I would need to introduce a RestConsumer parallel to the RPCConsumer. Maybe with a common superclass Consumer to handle errors? It's a lot of classes only for one specific use-case, do you think it's acceptable? Having no call limit is definitely an important upside!

Reply:
You're right, it's little bit too much for now, and the call limit of 30 is not a big deal for now, but let's keep this idea for later, maybe create an issue for a future work.

But we still need to handle when the `@username` is not resolve and the `rpc.nano.to` respond an error.

Reply:
For sure! I can create an issue :)

When `@username` is not resolved, a NonaParseError is thrown. This is because `NameInfo` expectes `key` to be present - in the error case `error` and `message` are return, as tested in `should not resolve an invalid username` of `name-service.test.ts`. I tried to follow the schema of your existing tests, so please let me know how exactly you want these methods tested!

Reply:
The problem isn't with the test, don't worry about that, the problem is with the user, they won't understand the error.
It would therefore be preferable to handle the rpc.nano.to error and respond with the correct message.

If you think that this PR is already too important, we can do it in another one, it's just a question of having a better error message.

Reply:
I added an explicit error to this case, is that what you had in mind?

Reply:
Not really, sorry I'm not clear.
This error handling must be in the Rpc class, because rpc.nano.to returns an error message which is different from a classic rpc message.

If you test this code :
```
  const nona = new Nona();
  await nona.nameService.resolveTarget(‘@not-a-real-user’);
```
You'll see that `data` retrieved from line 24 in `rpc.ts` is equal to `{ error: 500, message: ‘Username @not-a-real-user not registered.’ }`
Whereas a node error message is formatted like this: `{ error: string }` and so the `handleResponseDataError` method on line 39 in `rpc.ts` does not validate the if on line 40 because `error` is a `number` and not a `string`. This results in a message with no information.
We need to manage this error format by passing the `data` object to `handleResponseDataError` and manage the case of a classic node error and a `rpc.nano.to` error.

To have something like this : 
```
private handleResponseDataError(errorData: Record<string, unknown>): never {
    if (typeof errorData.error === 'string') {
      throw new NonaRpcError(errorData.error);
    }
    if (typeof errorData.error === 'number' && typeof errorData.message === 'string') {
      throw new NonaRpcError(errorData.message);
    }

    throw new NonaRpcError('Unknown error occurred from RPC call');
  }
```

Sorry to bother you with this, if you want you can delete your last commit, I merge this PR and I create a new PR handling this case.

Reply:
You are in no way bothering me! I'm just trying to understand the requirements.

Do the current changes look good to you?

Reply:
Perfect sound good to me !